Mathlib/RingTheory/Valuation/Integers.lean

```diff
@@ -133,6 +138,34 @@ theorem le_iff_dvd {x y : O} : v (algebraMap O F x) ≤ v (algebraMap O F y) ↔
   ⟨hv.dvd_of_le, hv.le_of_dvd⟩
 #align valuation.integers.le_iff_dvd Valuation.Integers.le_iff_dvd
 
+/--
+This is the special case of `Valuation.Integers.isUnit_of_one` when the valuation is defined
+over a field. Let `v` be a valuation on some field `F` and `O` be its integers. For every element
+`x` in `O`, `x` is a unit in `O` if and only if the image of `x` in `F` has valuation 1.
+-/
+theorem isUnit_of_one' {x : O} (hvx : v (algebraMap O F x) = 1) :
+    IsUnit x := by
+  apply isUnit_of_one hv _ hvx
+  apply IsUnit.mk0
+  rw [← v.ne_zero_iff]
+  simp only [hvx, ne_eq, one_ne_zero, not_false_eq_true]
+
+theorem eq_algebraMap_or_inv_eq_algebraMap (x : F) :
+    ∃ a : O, x = algebraMap O F a ∨ x⁻¹ = algebraMap O F a := by
+  by_cases h : v x ≤ 1
+  · obtain ⟨a, ha⟩ := exists_of_le_one hv h
+    use a
+    exact Or.inl ha.symm
+  · have : v x⁻¹ ≤ 1 := by
+      apply le_of_lt
+      push_neg at h
+      rwa [← Valuation.one_lt_val_iff]
+      intro h0
+      simp [h0] at h
```

Comment:
We should have `Valuation.val_le_one_iff`

Reply:
Yes! I'll do this.

Should we have four lemmas? `one_lt_val_iff`, `one_le_val_iff`, `val_le_one_iff`, `val_lt_one_iff`? Or only two of them?

Reply:
Hmmm. Or maybe not and we should just `rw [map_inv₀, inv_le_one₀]`

Reply:
I added four lemmas and the following one
```
-- This theorem is not stated in the strongest form but is very useful.
theorem val_le_one_or_val_inv_le_one (v : Valuation K Γ₀) (x : K) : v x ≤ 1 ∨ v x⁻¹ ≤ 1 := by
```
and used it to golf the `Valuation.valuationSubring`